tests/foreman/cli/test_activationkey.py

```diff
@@ -1256,3 +1258,29 @@ def test_positive_content_override(self):
                     u'organization-id': self.org['id'],
                 })
                 self.assertEqual(content[0]['enabled?'], override_value)
+
+    @tier2
+    def test_positive_remove_user(self):
+        """Delete any user who has previously created an activation key
+        and check that activation key still exists
+
+        @id: ba9c4b29-2349-47ea-8081-917de2c17ed2
+
+        @Assert: Activation Key can be read
+
+        @BZ: 1291271
+        """
+        password = gen_string('alpha')
+        user = make_user({'password': password, 'admin': 'true'})
+        ak = ActivationKey.with_user(
+            username=user['login'],
+            password=password
+        ).create({
+            'name': gen_string('alpha'),
+            'organization-id': self.org['id'],
+        })
+        User.delete({'id': user['id']})
+        try:
+            ActivationKey.info({'id': ak['id']})
+        except CLIReturnCodeError:
```

Comment:
Is it certain that searching a non-existent id will always end up with this error? For example on Sat 6.2 cli simply returns help in such case. Wouldn't it be safer to assert that the id is present in the list of activation keys?